packages/graphql/src/translate/cypher-builder/statements/Match.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CypherContext } from "../CypherContext";
+import { MatchableElement, MatchParams, MatchPattern } from "../MatchPattern";
+import { Node } from "../references/Node";
+import { Param } from "../references/Param";
+import { Query } from "./Query";
+import { ReturnStatement } from "./Return";
+
+type Params = Record<string, Param<any>>;
+
+type Where = Map<MatchableElement, Params>;
+
+export class Match<T extends MatchableElement> extends Query {
+    private matchPattern: MatchPattern<T>;
+    private whereParams: Where;
+
+    // parameters cast required due to neo-push
```

Comment:
Hold on... We're having to add changes to our _production code_... Because of `neo-push`?!